src/types.ts

```diff
@@ -0,0 +1,31 @@
+import type { JsonRpcRequest } from 'json-rpc-engine';
+
+export type RequestHeaders = Record<string, string>;
+
+export type ExtendedJsonRpcRequest<T> = JsonRpcRequest<T> & { origin?: string };
+
+/**
+ * These are networks:
+ *
+ * 1. for which Infura has released official, production support ({@see {@link https://docs.infura.io}})
+ * 2. which support the JSON-RPC 2.0 protocol
+ */
+export type InfuraJsonRpcSupportedNetwork =
```

Comment:
Not sure if it's worth listing *all* of the possible networks Infura takes, as we don't use them. However there shouldn't be anything in this list for which JSON-RPC is not supported.